manifests/central_server.pp

```diff
@@ -3,64 +3,72 @@
   path => [
     '/usr/local/bin',
     '/opt/local/bin',
+    '/opt/ejabberd/bin',
     '/usr/bin',
     '/usr/sbin',
     '/bin',
     '/sbin'],
   logoutput => true,
   }
-
-  class { 'ejabberd':
-    config_content  => template('archipel/ejabberd.cfg.erb'),
-    package_ensure  => 'installed',
-    package_name    => 'ejabberd',
-    service_reload  => true,
+  include archipel
+  package { "expect":
+    ensure      => installed
   }
-
-
-  package { ['erlang-xmlrpc','erlang-tools','erlang-xmerl','subversion']:
-    ensure => present,
-    require => Package['ejabberd'],
+  ->
+  exec { "retrieve_ejabberd_binary_installer ":
+    command     => "wget $ejabberd_download_url -P /usr/local/bin",
+    creates     => "/usr/local/bin/$ejabberd_binary_installer",
+    tries       => 2,
+    timeout     => 900  #set the timeout to be 15 min
   }
   ->
-  vcsrepo { '/usr/local/src/ejabberd-modules':
-    ensure      => present,
-    provider    => svn,
-    source      => 'http://svn.process-one.net/ejabberd-modules/',
+  file { "ejabberd binary installer":
+    path        => "/usr/local/bin/$ejabberd_binary_installer",
+    owner       => root,
+    mode        => 0755
   }
   ->
-  exec { "compile-ejabberd-xmlrpc":
-    cwd         => "/usr/local/src/ejabberd-modules/ejabberd_xmlrpc/trunk/",
-    command     => "/usr/local/src/ejabberd-modules/ejabberd_xmlrpc/trunk/build.sh",
-    creates     => "/usr/local/src/ejabberd-modules/ejabberd_xmlrpc/trunk/ebin/mod_xmlrpc.beam",
-    environment => 'HOME=/root',
-    logoutput   => true,
+  file { "fetching the binary installer expect script":
+    path        => "/usr/local/bin/ejabberd_binary_install.exp",
+    content     => template('archipel/ejabberd_binary_install.exp.erb'),
+    owner       => root,
+    mode        => 0755
   }
   ->
-  file { "${ejabberd::params::lib_dir}/ebin/ejabberd_xmlrpc.beam":
-    ensure  => present,
-    source  => "/usr/local/src/ejabberd-modules/ejabberd_xmlrpc/trunk/ebin/ejabberd_xmlrpc.beam",
+  exec { "running the binary installer via expect script ":
+    command     => "expect /usr/local/bin/ejabberd_binary_install.exp",
+    user        => root,
+    timeout     => 0  #disable the timeout for binary installer
```

Comment:
Create a user to run ejabberd as non privilegied user (home located to /opt/ejabberd).
Install the binary as the ejabberd user ejabberd-15.0X-linux-x86_64-installer.run --mode unattended --adminpw admin --prefix /opt/ejabberd/